core/src/main/java/com/datastax/oss/driver/internal/core/DefaultCluster.java

```diff
@@ -145,25 +149,63 @@ private void init() {
           .addContactPoints(initialContactPoints)
           .thenCompose(v -> context.topologyMonitor().init())
           .thenCompose(v -> metadataManager.refreshNodes())
-          .thenAccept(
-              v -> {
-                try {
-                  context.loadBalancingPolicyWrapper().init();
-                  context.configLoader().onDriverInit(context);
-                  LOG.debug("[{}] Initialization complete, ready", logPrefix);
-                  initFuture.complete(DefaultCluster.this);
-                  // TODO schedule full schema refresh asynchronously (does not block init)
-                } catch (Throwable throwable) {
-                  initFuture.completeExceptionally(throwable);
-                }
-              })
+          .thenAccept(this::afterInitialNodeListRefresh)
           .exceptionally(
               error -> {
                 initFuture.completeExceptionally(error);
                 return null;
               });
     }
 
+    private void afterInitialNodeListRefresh(@SuppressWarnings("unused") Void ignored) {
+      try {
+        boolean protocolWasForced =
+            context.config().getDefaultProfile().isDefined(CoreDriverOption.PROTOCOL_VERSION);
+        boolean needSchemaRefresh = true;
+        if (!protocolWasForced) {
+          ProtocolVersion currentVersion = context.protocolVersion();
+          ProtocolVersion bestVersion =
+              context
+                  .protocolVersionRegistry()
+                  .highestCommon(metadataManager.getMetadata().getNodes().values());
```

Comment:
Should we consider only non-ignored nodes?  Use case I can think of is if you have another DC that hasn't been upgraded yet.   You won't connect to that DC, so you might not want to consider its supported protocol versions. 

One use case is if you are migrating your infra to another cloud provider and brought up another DC with a newer version that you are connecting to but keeping up the old DC in the meantime.  That is probably an edge case, and even in that event you probably don't want to be using a new protocol version anyways.